.github/actions/verify-badge/action.yaml

```diff
@@ -0,0 +1,257 @@
+
+        NETPOL_EXT_SELECTOR_COUNT=$(echo "$UDS_PACKAGE_JSON" | jq -r '.spec.network.allow[] | select(.remoteGenerated != "IntraNamespace") | select(.selector != null) | length' | wc -l)
+        echo "ℹ️ Non-IntraNamespace network policies with selectors: $NETPOL_EXT_SELECTOR_COUNT"
+
+        if [ "$NETPOL_EXT_COUNT" -ne "$NETPOL_EXT_SELECTOR_COUNT" ]; then
+          echo "❌ Not all applicable network policies are using selectors"
+          exit 1
+        fi
+        echo "✅ All applicable network policies are using selectors"
+
+        NETPOL_EXT_NOTKUBE_COUNT=$(echo "$UDS_PACKAGE_JSON" | jq -r '.spec.network.allow[] | select(.remoteGenerated != "IntraNamespace") | select(.remoteGenerated != "KubeAPI") | length' | wc -l)
+        echo "ℹ️ Non-IntraNamespace, non-KubeAPI network policies: $NETPOL_EXT_NOTKUBE_COUNT"
+
+        NETPOL_EXT_NOTKUBE_PORT_COUNT=$(echo "$UDS_PACKAGE_JSON" | jq -r '.spec.network.allow[] | select(.remoteGenerated != "IntraNamespace") | select(.remoteGenerated != "KubeAPI") | select(.ports != null or .port != null) | length' | wc -l)
+        echo "ℹ️ Non-IntraNamespace, non-KubeAPI network policies with ports: $NETPOL_EXT_NOTKUBE_PORT_COUNT"
+
+        if [ "$NETPOL_EXT_NOTKUBE_COUNT" -ne "$NETPOL_EXT_NOTKUBE_PORT_COUNT" ]; then
+          echo "❌ Not all applicable network policies are using ports"
+          exit 1
+        fi
+        echo "✅ All applicable network policies are using ports"
+
+        echo
+
+        # Keycloak
+
+        SSO_CLIENT_COUNT=$(echo "$UDS_PACKAGE_JSON" |  jq -r '.spec.sso | length')
+
+        ## Must use and create a Keycloak client through the sso key if the application provides a user login.
+        if [ "$SSO_CLIENT_COUNT" -gt 0 ]; then
+          echo "ℹ️ There are $SSO_CLIENT_COUNT SSO clients defined"
+
+          # Should consider security options during implementation to provide the most secure default possible (i.e. SAML w/SCIM vs OIDC).
+          SSO_CLIENT_PROTOCOL=$(echo "$UDS_PACKAGE_JSON" | jq -r '.spec.sso[].protocol')
+          if [ "$SSO_CLIENT_PROTOCOL" != "saml" ]; then
+            echo "❌ SAML is not the default protocol"
+            exit 1
+          fi
+          echo "✅ Default protocol is SAML"
+
+          # Should name the client <App> Login (i.e. Mattermost Login) to provide login UX consistency.
+          SSO_CLIENT_NAME=$(echo "$UDS_PACKAGE_JSON" | jq -r '.spec.sso[].name')
+          if [ "$SSO_CLIENT_NAME" != "${UDS_PACKAGE_NAME^} Login" ]; then
+            echo "❌ SSO client name not in correct format"
+            exit 1
+          fi
+          echo "✅ SSO client name is correct"
+
+          # Should clearly mark the client id with the group and app name uds-<group>-<application> (i.e. uds-swf-mattermost) to provide consistency in the Keycloak UI.
+          SSO_CLIENT_ID=$(echo "$UDS_PACKAGE_JSON" | jq -r '.spec.sso[].clientId')
+          if [ "$SSO_CLIENT_ID" != "uds-package-${UDS_PACKAGE_NAME}-${SSO_CLIENT_PROTOCOL}" ]; then
```

Comment:
Package is not in the example format, may want to strike that and allow a regex or similar for that piece (i.e. could be SWF/LFAI/something else).

Reply:
Annoying Nexus question. Its current client id is "uds:nexus" because it has to be a URI. Can we update this to be able to handle such a client ID.

Reply:
@anthonywendt I think since this is a `should` it should be modified to just warn (not fail), which would cover Nexus and other apps that might require naming differently.

Reply:
added an input and changed to a warning